lib/faye/redis.rb

```diff
@@ -205,11 +210,16 @@ def redis
           @server.info "Faye::Redis: redis connection reconnected"
         end
         connection.on(:reconnect_failed) do |count|
-          @server.info "Faye::Redis: redis connection reconnect failed (#{count}/4)"
+          @server.info "Faye::Redis: redis connection reconnect failed #{count} time(s)"
+          begin
+            fn = @options[:reconnect_failed] || ->(_, _) {}
+            fn.call(count, 'redis')
+          rescue => e
+            @server.error "Faye::Redis: Execution of reconnect_failed lambda failed with #{e} (redis connection)"
```

Comment:
`reconnect_failed` is passed by the API user so we want to safely execute the code.